R/deployApp.R

```diff
@@ -351,6 +353,20 @@ deployApp <- function(appDir = getwd(),
     application <- applicationForTarget(client, accountDetails, target, forceUpdate)
   })
 
+  # Shinyapps defaults to public visibility.
+  # Other values should be set before data is deployed.
+  currentVisibility <- application$deployment$properties$application.visibility
```

Comment:
Move computation of `currentVisibility` into your `isShinyapps` block. Only shinyapps has that data structure.